internal/ir/ratelimit_infra.go

```diff
@@ -5,13 +5,17 @@
 
 package ir
 
+import (
+	"github.com/envoyproxy/gateway/api/config/v1alpha1"
+)
+
 // RateLimitInfra defines managed rate limit service infrastructure.
 // +k8s:deepcopy-gen=true
 type RateLimitInfra struct {
-	// Rate limit service configuration
-	Configs []*RateLimitServiceConfig
-	// Backend holds configuration associated with the backend database.
-	Backend *RateLimitDBBackend
+	// ServiceConfigs for Rate limit service configuration.
+	ServiceConfigs []*RateLimitServiceConfig
+	// Config ratelimit configuration.
+	Config *v1alpha1.EnvoyGateway
```

Comment:
do we need to pass entire EnvoyGateway config here, or just  `rateLimitDeployment` ?

Reply:
since EnvoyGateway is a startup config, it exists in every runner, so maybe https://github.com/envoyproxy/gateway/blob/6fc9f8dfe8883df61343b133da694257eb313f69/internal/infrastructure/kubernetes/infra.go#L19 can be enhanced to store envoy gateway config

Reply:
> do we need to pass entire EnvoyGateway config here, or just `rateLimitDeployment` ?

using `EnvoyGateway` is convinced for other extensions, like supporting different `Backend`.

Reply:
> since EnvoyGateway is a startup config, it exists in every runner, so maybe
> 
> https://github.com/envoyproxy/gateway/blob/6fc9f8dfe8883df61343b133da694257eb313f69/internal/infrastructure/kubernetes/infra.go#L19
> 
> can be enhanced to store envoy gateway config

store envoy gateway config to `internal/infrastructure/kubernetes/infra.go:27`, and assign the config when the subscription changes.